File: src/day-1/day-1-pt-2.ts
import { INPUT } from './input';

let currentSumOfCalories = 0;

const calorieTotals: number[] = INPUT.split('\n').reduce(
  (acc: number[], currentFoodCaloriesStr: string) => {
    if (!currentFoodCaloriesStr) {
      acc.push(currentSumOfCalories);
      currentSumOfCalories = 0;
      return acc;
    }

    currentSumOfCalories += Number.parseInt(currentFoodCaloriesStr.trim());
    return acc;
  },
  []
);

const topThreeTotalCalories = calorieTotals
  .sort((a, b) => a - b)
  .slice(-3)
  .reduce((sum, calories) => sum + calories);
console.log(topThreeTotalCalories);

File: python/day_1/day_1_pt_2.py
current_sum_of_calories = 0
totals = []

with open('input.txt', encoding="utf-8") as input:
    for line in input:
        calories = line.strip()

        if (calories == ''):
            totals.append(current_sum_of_calories)
            current_sum_of_calories = 0
        else:
            current_sum_of_calories += int(calories)

totals.sort()
print(sum(totals[-3:]))